crates/torii/core/src/utils.rs

```diff
@@ -46,6 +48,18 @@ pub async fn fetch_content_from_ipfs(cid: &str, mut retries: u8) -> Result<Bytes
         IPFS_CLIENT_MAX_RETRY, cid
     )))
 }
+
+pub async fn health_check_provider<P: Provider + Sync + std::fmt::Debug + 'static>(
+    provider: Arc<P>,
```

Comment:
```suggestion
    provider: P,
```
We don't need an `Arc` here since `Arc<P>` already implements `Provider`.

Reply:
fixed.